server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/PreprocessorStack.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.preprocessor.delegates;
+
+import lombok.NonNull;
+
+import java.util.Deque;
+import java.util.Objects;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * Provides a contract and default implementation of various stack operation. To be used while
+ * assembling the {@link org.eclipse.lsp.cobol.core.model.ExtendedDocument} from pre-processors.
+ */
+public interface PreprocessorStack {
```

Comment:
Why not have an abstract class?

Reply:
No preference here. Made it interface out of laziness 🥱 
Listeners already extend a class. 
Do you suggest switching to abstract class?

Reply:
I think it's better to keep it being an interface